Sming/Libraries/OtaUpgrade/OtaChecksumVerifier.h

```diff
@@ -0,0 +1,49 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2017 by Slavey Karadzhov
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * OtaChecksumVerifier.h
+ *
+ ****/
+
+#pragma once
+
+#include "romfunc_md5.h"
+#include <string.h>
+
+/**
+ * @brief Checksum verifier used by `BasicOtaUpgradeStream` if signature verification is disabled.
+ *
+ * This is a simple C++ wrapper for the MD5 ROM functions.
+ */
+class OtaChecksumVerifier
+{
+	MD5Context context;
+
+public:
+	static const size_t NumBytes = MD5_SIZE; ///< Size of checksum in bytes.
+
+	OtaChecksumVerifier()
+	{
+		MD5Init(&context);
+	}
+
+	/** Continue MD5 checksum calculation for the given chunk of data.
+	 */
+	void update(const void* data, size_t size)
+	{
+		MD5Update(&context, data, size);
+	}
+
+	/** Verify given \c checksum.
+	 * @return `true` if checksum and content matches, `false` otherwise.
+	 */
+	bool verify(const uint8_t (&checksum)[MD5_SIZE])
```

Comment:
Can you please point me to the header where `Data` is defined? It somehow works, but neither I nor the IDE can locate the definition.

Reply:
For `Data` see comments in `OtaChecksumVerifier.h`.
NB. Are you compiling with `ENABLE_OTA_SIGNING=0`?

Reply:
```
class OtaChecksumVerifier
{
public:
	struct Data {
		uint8_t content[MD5_SIZE];

		bool operator==(const Data& other) const
		{
			return memcmp(content, other.content, sizeof(content)) == 0;
		}
	};

	OtaChecksumVerifier()
	{
		MD5Init(&context);
	}

	/** Continue MD5 checksum calculation for the given chunk of data.
	 */
	void update(const void* data, size_t size)
	{
		MD5Update(&context, data, size);
	}

	/** Verify given \c checksum.
	 * @return `true` if checksum and content matches, `false` otherwise.
	 */
	bool verify(const Data& checksum)
	{
		Data expectedChecksum;
		MD5Final(expectedChecksum.content, &context);
		return checksum == expectedChecksum;
	}

private:
	MD5Context context;
};
```

Reply:
Sorry for the confusion above, I should have scrolled down first.

I understand your intend by creating a `Data` class to encapsulate the array. However, in this case it creates way too much boilerplate code for my taste. Luckily, there is `std::array`, which can be used instead.

Reply:
std library, even better!